### PR TITLE
move admission read logic to apiserver

### PR DESCRIPTION
**Pull request by deads2k**
Promised followup to https://github.com/kubernetes/kubernetes/pull/40943

@sttts @kubernetes/sig-api-machinery-misc

### CODE REVIEW

**Comment by deads2k** [comment]
@eparis @apelisse are changes to `vendor/BUILD` always top level?  Our staging directories hit those.  Should I create an owners file in there or can special case and exclude it?  It's a generated file.

**Comment by apelisse** [comment]
@deads2k Yeah, we're kind of trying to figure this out at the moment ...

**Comment by deads2k** [comment]
@k8s-bot gce etcd3 e2e test this

**Comment by deads2k** [comment]
@k8s-bot gce etcd3 e2e test this

File: staging/src/k8s.io/apiserver/pkg/apis/apiserver/doc.go [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+// +k8s:deepcopy-gen=package,register
+
+// Package api is the internal version of the API.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/apis/apiserver/doc.go]: comment is wrong

File: staging/src/k8s.io/apiserver/pkg/apis/register.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+)
+
+// GroupFactoryRegistry is the APIGroupFactoryRegistry (overlaps a bit with Registry, see comments in package for details)
+var GroupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+
+// Registry is an instance of an API registry.  This is an interim step to start removing the idea of a global
+// API registry.
+var Registry = registered.NewOrDie(os.Getenv("KUBE_API_VERSIONS"))
+
+// Scheme is the default instance of runtime.Scheme to which types in the Kubernetes API are already registered.
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/apis/register.go]: comment is wrong

File: pkg/kubeapiserver/admission/config.go [hunk 3]
```diff
@@ -31,9 +31,11 @@ import (
...
-	"k8s.io/kubernetes/pkg/api"
-	"k8s.io/kubernetes/pkg/apis/componentconfig"
-	componentconfigv1alpha1 "k8s.io/kubernetes/pkg/apis/componentconfig/v1alpha1"
+	"k8s.io/apiserver/pkg/apis"
+	"k8s.io/apiserver/pkg/apis/apiserver"
+	apiserverv1alpha1 "k8s.io/apiserver/pkg/apis/apiserver/v1alpha1"
+
+	_ "k8s.io/apiserver/pkg/apis/apiserver/install"
 
 	runtime "k8s.io/apimachinery/pkg/runtime"
```
Comment (sttts) [inline pkg/kubeapiserver/admission/config.go]: move up

File: staging/src/k8s.io/apiserver/pkg/admission/config.go [hunk 4]
```diff
@@ -29,17 +29,27 @@ import (
...
 	apiserverv1alpha1 "k8s.io/apiserver/pkg/apis/apiserver/v1alpha1"
+)
 
-	_ "k8s.io/apiserver/pkg/apis/apiserver/install"
-
-	runtime "k8s.io/apimachinery/pkg/runtime"
+var (
+	groupFactoryRegistry = make(announced.APIGroupFactoryRegistry)
+	registry             = registered.NewOrDie(os.Getenv("KUBE_API_VERSIONS"))
+	scheme               = runtime.NewScheme()
```
Comment (sttts) [inline staging/src/k8s.io/apiserver/pkg/admission/config.go]: much better having it privately here 👍

**Comment by sttts** [comment]
some nits

**Comment by sttts** [comment]
Otherwise, lgtm.

**Comment by deads2k** [comment]
@k8s-bot node e2e test this

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this
@k8s-bot gce etcd3 e2e test this

**Comment by deads2k** [comment]
@k8s-bot cvm gce e2e test this

**Pull request merged**